src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1026,6 +1027,56 @@ def list_roles(self) -> List[Dict[str, Any]]:
 
         return roles
 
+    def create_role(self, role_name: str, role_path: str, role_assume_policy_doc: str) -> None:
+        try:
+            json.loads(role_assume_policy_doc)
+        except:  # noqa: E722
+            raise DashboardException('Assume role policy document is not a valid json')
+
+        def between(v, a, b):
+            return ord(v) >= ord(a) and ord(v) <= ord(b)
+        # valid values:
+        # pylint: disable=C0301
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-path # noqa: E501
+        valid_special = '_+=,.@'
+        if len(role_name) > 64:
+            raise DashboardException(
+                f'Role name "{role_name}" is invalid. Should be 64 characters or less')
+        for c in role_name:
+            if not (between(c, 'a', 'z')
+                    or between(c, 'A', 'Z')
+                    or between(c, '0', '9')
+                    or c in valid_special):
+                raise DashboardException(
+                    f'Role name "{role_name}" is invalid. Valid characters are [a-zA-Z_+=,.@]')
+
+        if not os.path.isabs(role_path):
+            raise DashboardException(
+                f'Role path "{role_path}" is invalid. It should be an absolute path')
+        if role_path[-1] != '/':
+            raise DashboardException(
+                f'Role path "{role_path}" is invalid. It should start and end with a slash')
+        for c in role_path:
+            if ord(c) > 0x7F or ord(c) < 0x21:
```

Comment:
No need to convert things from unicode (the standard Python 3 string representation) to integers. Additionally, unicode characters can be represented by names, which is friendlier than hex data: 
```suggestion
            if c => "\N{TILDE}" or c < "\N{EXCLAMATION MARK}":
```

I was trying to understand if there's a [Unicode Category](https://www.compart.com/en/unicode/category) that matches this range, but it doesn't seem so. In any case, the AWS spec is giving you a unicode regex pattern ready to use (`(\u002F)|(\u002F[\u0021-\u007E]+\u002F)`), so why not using that directly?